api/src/main/java/ai/djl/ndarray/BaseNDManager.java

```diff
@@ -404,9 +404,18 @@ NDManager getAlternativeManager() {
      * @throws IllegalArgumentException if buffer size is invalid
      */
     public static void validateBufferSize(Buffer buffer, DataType dataType, int expected) {
+        boolean isByteBuffer = buffer instanceof ByteBuffer;
+        DataType type = DataType.fromBuffer(buffer);
+        if (!isByteBuffer && type != dataType) {
```

Comment:
The following mismatch case will escape the check:
`type != dataType && buffer is ByteBuffer`, but `dataType` is not one of the byte types = {DataType.UINT8, DataType.INT8, DataType.BOOLEAN}

```suggestion

if (arrayType != inputType ) {
          DataType[] types = {DataType.UINT8, DataType.INT8, DataType.BOOLEAN};
          if (!isByteBuffer || Arrays.stream(types).noneMatch(x -> x == dataTypeType)) {
          
```

Reply:
ByteBuffer should always be allowed. It's the memory representation of all data types. All NDArray has `toByteBuffer()` method. The JNI only accept ByteBuffer, if we only accept matching Buffer, we have to covert ByteBuffer to Buffer and copy the Buffer to ByteBuffer to pass to JNI. It's not efficient for Hybrid engine. We trying to achieve 0 copy between pytorch and onnx engines. It rely on ByteBuffer.